rimage/image_file.go

```diff
@@ -71,12 +81,48 @@ func init() {
 			}, nil
 		},
 	)
-}
+
+	// Here we register our format for depth images so that we can use
+	// image.Decode as long as we have the appropriate header
+	image.RegisterFormat("vnd.viam.dep", string(DepthMapMagicNumber),
+		func(r io.Reader) (image.Image, error) {
+			f := r.(*bufio.Reader)
+			dm, err := ReadDepthMap(f)
+			if err != nil {
+				return nil, err
+			}
+			return dm, nil
+		},
+		func(r io.Reader) (image.Config, error) {
+			// We cannot read the height and width from bytes b/c we don't know (yet)
+			// how the depth map is encoded, so send it to decode and then grab it.
+			// Using Gray 16 as underlying color model for depth
+			f := r.(*bufio.Reader)
+			firstBytes, err := _readNext(r)
+			if err != nil || firstBytes != MagicNumIntViamType {
+				return image.Config{}, err
```

Comment:
If the first bytes != the Magic Number, create an error message saying as much, and return that error.